nni/mutable/mutable.py

```diff
@@ -501,3 +524,608 @@ class MutableSymbol(LabeledMutable, Symbol, MutableExpression):
+        """Iterate over all possible values.
+
+        If ``n_chosen`` is None, iterate over all possible subsets, in the order of increasing length.
+        Otherwise, iterate over all possible combinations of ``n_chosen`` length,
+        using the implementation of :func:`itertools.combinations`.
+
+        See Also
+        --------
+        Mutable.grid
+        """
+        memo = {} if memo is None else memo
+        err = self.check_contains(memo)
+
+        if isinstance(err, SampleMissingError):
+            if self.n_chosen is not None:
+                gen = itertools.combinations(self.values, self.n_chosen)
+            else:
+                gen = itertools.chain.from_iterable(itertools.combinations(self.values, r) for r in range(len(self.values) + 1))
+
+            assert self.label not in memo, 'Memo should not contain the label.'
+            for value in gen:
+                memo[self.label] = list(value)
+                yield self.freeze(memo)
+            memo.pop(self.label)
+        else:
+            yield self.freeze(memo)
+
+
+class Continuous(MutableSymbol):
+    """A variable from a continuous numeric domain.
+
+    It supports most commonly used distributions including uniform, loguniform,
+    normal, lognormal, as well as the quantized version.
+    It also supports using arbitrary distribution from :mod:`scipy.stats`.
+
+    Parameters
+    ----------
+    low
+        The lower bound of the domain. Used for uniform and loguniform.
+        It will also be used to clip the value if it is outside the domain.
+    high
+        The upper bound of the domain. Used for uniform and loguniform.
+        It will also be used to clip the value if it is outside the domain.
+    mu
+        The mean of the domain. Used for normal and lognormal.
+    sigma
+        The standard deviation of the domain. Used for normal and lognormal.
+    log_distributed
+        Whether the domain is log distributed.
+    quantize
+        If specified, the final value will be postprocessed with
+        ``clip(round(uniform(low, high) / q) * q, low, high)``,
+        where the clip operation is used to constrain the generated value within the bounds.
+        For example, when quantize is 2.5, all the values will be rounded to the nearest multiple of 2.5.
+        Note that, if ``low`` or ``high`` is not a multiple of ``quantize``,
+        it will be clipped to ``low`` or ``high`` **after** rounding.
+    distribution
+        The distribution to use. It should be a ``rv_frozen`` instance,
+        which can be obtained by calling ``scipy.stats.distribution_name(...)``.
+        If specified, ``low``, ``high``, ``mu``, ``sigma``, ``log_distributed`` will be ignored.
+    default
+        The default value. If not specified, the default value will be the median of distribution.
+    label
+        The label of the variable.
+
+    Examples
+    --------
+    To create a variable uniformly sampled from 0 to 1::
```

Comment:
I don't think it's a good idea to let the distribution (e.g. normal, uniform) to be induced from whether `low, high` or `mu, sigma` is given. I prefer to let `Continuous` to be the base class (or with a default `uniform` random generator), and let `normal`, `lognormal` to have separate class.

Reply:
uniform and normal are special cases of Continuous. They will be subclasses of Continuous.

The idea of Continuous is to support arbitrary distributions. Normal and uniform are only some shortcuts here.